src/sql/workbench/contrib/executionPlan/browser/executionPlanComparisonEditorView.ts

```diff
@@ -392,8 +395,14 @@ export class ExecutionPlanComparisonEditorView {
 				if (this._topPlanDiagramModels && this._bottomPlanDiagramModels) {
 					this._topPlanDiagramModels[this._activeTopPlanIndex].graphFile.graphFileType = 'sqlplan';
 					this._bottomPlanDiagramModels[this._activeBottomPlanIndex].graphFile.graphFileType = 'sqlplan';
+
+					const semaphore = generateUuid();
+					this._currentRequestUuid = semaphore;
 					const result = await this._executionPlanService.compareExecutionPlanGraph(this._topPlanDiagramModels[this._activeTopPlanIndex].graphFile,
 						this._bottomPlanDiagramModels[this._activeBottomPlanIndex].graphFile);
+					if (semaphore !== this._currentRequestUuid) {
```

Comment:
Wouldn't this always resolve to false because the semaphore and currentRequestUuid are always the same?

Reply:
The purpose of this code is to make sure that we discard the older request that might be running and not completed.
Suppose, we have a  request to compare plans that is taking a long time. The semaphore and this._currentRequestUuid is 'req1'.

While this is running, the user switches the graph and now we are comparing 2 new graphs. The request id for this semaphore and this._currentRequestUuid is 'req2'. 

If the older request 'req1' returns back some result, we will discard it because this._currentRequestUuid (req2) and semaphore (req1) value won't match.

Reply:
The purpose of this code is to make sure that we discard the older request that might be running and not completed.
Suppose, we have 1 request to compare plans that i. The semaphore and  this._currentRequestUuid is 'req1'.

While this is running, the user switches the graph and now we are comparing 2 new graphs. The request id for this semaphore and this._currentRequestUuid is 'req2'.

Reply:
I guess, I can change the name because this is not exactly a semaphore. Initially, I thought of using one but that did not work out.